src/Cassette/BundleCollection.cs

```diff
@@ -54,6 +76,23 @@ internal Exception InitializationException
                 using (GetWriteLock())
                 {
                     initializationException = value;
+
+                    if (value == null)
+                    {
+                        // reset our collected exceptions
+                        errorCollectingException = null;
+                        return;
+                    }
+
+                    if (errorCollectingException == null) errorCollectingException = new ErrorCollectingException();
+
+                    errorCollectingException.Exceptions.Add(value);
+                    if (errorCollectingException.Exceptions.Count > 32 || value is ThreadAbortException)

```

Comment:
We should add comments to explain the rationale behind the changes for future maintainability